src/utils/request.js

```diff
@@ -1,54 +1,32 @@
 import { extend } from 'umi-request';
-import { notification } from 'antd';
-import router from 'umi/router';
-
-const codeMessage = {
-  200: 'The server successfully returned the requested data. ',
-  201: 'New or modified data is successful. ',
-  202: 'A request has entered the background queue (asynchronous task). ',
-  204: 'The data was deleted successfully. ',
-  400: 'The request returned an error and the server did not perform any new or modified data operations. ',
-  401: 'User does not have permission (token, username, or password is incorrect). ',
-  403: 'The user is authorized, but access is forbidden. ',
-  404: 'The request was made for a record that does not exist. ',
-  406: 'The format of the request is not available. ',
-  410: 'The requested resource is permanently deleted and will not be retrieved. ',
-  422: 'A validation error occurred when creating an object. ',
-  500: 'An error occurred on the server. Please check the server. ',
-  502: 'Gateway error. ',
-  503: 'The service is unavailable and the server is temporarily overloaded or maintained. ',
-  504: 'The gateway timed out. ',
-};
 
 const errorHandler = error => {
-  const { response = {} } = error;
-  const errortext = codeMessage[response.status] || response.statusText;
-  const { status } = response;
-
-  notification.error({
-    message: `Request Error ${status}`,
-    description: errortext,
-  });
-
-  if (status === 403) {
-    router.push('/exception/403');
-    return;
-  }
-  if (status === 404) {
-    router.push('/exception/404');
-    return;
-  }
-  if (status <= 504 && status >= 500) {
-    router.push('/exception/500');
-    return;
-  }
-  if (status >= 405 && status < 422) {
-    router.push('/exception/404');
-  }
+  throw error;
 };
 
 const request = extend({
   errorHandler, // extend default error handler with custom actions
```

Comment:
Is this really necessary if we're just throwing the error we were passed anyway? What would happen without the errorHandler?

Reply:
Hey Dave, so if you navigate to 'src/models/auth' you'll find that we custom handle these errors in our effects. Yes, it is a breaking change which will affect all of our dashboard queries, so you're right in raising that concern

Reply:
Ah; right, the errors wouldn't normally be thrown, would they? Is that really the best way to handle this vs looking at HTTP status codes directly... aside from the issues with convincing umi to return the entire Response object, at least. But I suppose that was the motive here?

Reply:
We can forgo the `errorHandler` option completely as `umi-request` is based on the fetch and axios libraries and will allow you to capture errors with simple Promise chaining. We'll handle errors on a case by case on the component level by adopting best practices for error handling with Redux-Saga (https://redux-saga.js.org/docs/basics/ErrorHandling.html).